token-swap/proposals/Pausable.md

```diff
@@ -0,0 +1,19 @@
+# Pausable & Ownable 
+
+Implement shared library for SPL that can be used to extend contracts with ability to pause, resume and check for the owner before instructions are executed.
+
+PR should include:
+* Implementation of Ownable library with:
```

Comment:
We should probably view this as a program with the instructions you've listed:
* set owner
* renounce ownership
* check owner

Other parts:
* an `Owner` struct which contains `Option<Pubkey>`
* library code to generate a pda, probably given some `(struct_key, program_id)`
* test creating it from within another program and using it?